### ISSUE TITLE
[question] why stop_gradient on action taken w.r.t. categorical distribution but not other distribution

### ISSUE THREAD

**Issue by denyHell** [hill-a/stable-baselines#672]
In common/distribution, when computing neglogp, stop_gradient only applies to action taken w.r.t. categorical distribution but not other distribution. Shouldn't stop_gradient always be applied to the sampled actions as they are functions of network parameter? I am thinking about this in the context of implementing PPO2, where actions are sampled from the 'old policy'.

**Comment by araffin** [comment]
Hello,
the `stop_gradient` is applied here because of a change in Tensorflow api (see [doc](https://www.tensorflow.org/api_docs/python/tf/compat/v1/nn/softmax_cross_entropy_with_logits_v2)).
In practice, because the sampled actions are originally numpy arrays (the one from the old policy), we could remove it.
Those actions are not connected to the graph and no gradient will be backpropagated (only gradient for the current actions `self.logits` will be computed).
That's why it is not done for others distributions.

**Comment by denyHell** [comment]
Got it, thanks!
Another question: when using PPO2,  I use minibatch=1 in training and somehow the probability distribution never improve. Could you know any potential issue with minibatch=1? Thanks again!
